src/renderers/dom/fiber/__tests__/ReactDOMFiber-test.js

```diff
@@ -1117,5 +1117,38 @@ describe('ReactDOMFiber', () => {
           'to empty a container.',
       );
     });
+
+    it('should render a text component with a text DOM node on the same document as the container', () => {
+      // Arrange
+      // 1. Create a new document through the use of iframe
+      // 2. Set up the spy to make asserts when a text component
+      //    is rendered inside the iframe container
+      var textContent = 'Hello world';
+      var iframe = document.createElement('iframe');
+      document.body.appendChild(iframe);
+      var iframeDocument = iframe.contentDocument;
+      iframeDocument.write(
+        '<!DOCTYPE html><html><head></head><body><div></div></body></html>',
+      );
+      iframeDocument.close();
+      var iframeContainer = iframeDocument.body.firstChild;
+
+      var actualDocument;
+      var textNode;
+      iframeContainer.appendChildBackup = iframeContainer.appendChild;
+
+      spyOn(iframeContainer, 'appendChild').and.callFake(node => {
+        actualDocument = node.ownerDocument;
+        textNode = node;
+      });
+
+      // Act
```

Comment:
Nit: we can remove Arrange/Act/Assert comments as we don't use them in other tests.

Reply:
done :)

Reply:
I've also addressed the parallel structure comment from @sophiebits. Since now we have repeated code to do that same thing, I've also factored that out into a function.